src/Metadata/Driver/TypedPropertiesDriver.php

```diff
@@ -79,7 +85,12 @@ public function loadMetadataForClass(ReflectionClass $class): ?ClassMetadata
             try {
                 $propertyReflection = $this->getReflection($propertyMetadata);
                 if ($this->shouldTypeHint($propertyReflection)) {
-                    $propertyMetadata->setType($this->typeParser->parse($propertyReflection->getType()->getName()));
+                    $type = $propertyReflection->getType()->getName();
+                    if ('array' === $type) {
+                        $type = $this->docBlockTypeResolver->getPropertyDocblockTypeHint($propertyReflection);
```

Comment:
is there a case where `getPropertyDocblockTypeHint` is not able to resolve the type?

Reply:
Implementation has changed. Questions from the description are not current anymore, and this code is different.
DocBlockType resolver will return no type, if there is no `DocBlock` or `@var in DocBlock`